src/declarai/operators/base/types/operator.py

```diff
@@ -9,16 +9,17 @@
 
 class BaseOperator(ABC):
     llm: LLM
-    llm_params: Optional[Dict[str, Any]] = {}
 
-    def __init__(self, llm: LLM, parsed: PythonParser, **kwargs):
+    def __init__(self, llm: LLM, parsed: PythonParser, llm_params: Optional[Dict[str, Any]] = None, **kwargs):
         self.llm = llm
         self.parsed = parsed
         self.llm_response = None
+        self.llm_params = llm_params or {}
 
     @abstractmethod
     def compile(self, **kwargs) -> CompiledTemplate:
         ...
 
-    def predict(self, **kwargs) -> LLMResponse:
-        return self.llm.predict(**self.compile(**kwargs), **self.llm_params)
+    def predict(self, llm_params: Optional[Dict[str, Any]] = None, **kwargs) -> LLMResponse:
+        llm_params = llm_params or self.llm_params  # Order is important!
```

Comment:
I would rewrite the comment to say that provided params on execution have precedence of base llm-params defenitions

Reply:
Sure